src/main/java/com/mixpanel/android/viewcrawler/EditProtocol.java

```diff
@@ -90,35 +90,53 @@ public ViewVisitor readEdit(JSONObject source) throws BadInstructionsException {
-                targetClass = Class.forName(targetClassName);
-            } catch (final ClassNotFoundException e) {
-                throw new BadInstructionsException("Can't find class for visit path: " + targetClassName, e);
-            }
+                final Class<?> targetClass;
+                try {
+                    targetClass = Class.forName(targetClassName);
+                } catch (final ClassNotFoundException e) {
+                    throw new BadInstructionsException("Can't find class for visit path: " + targetClassName, e);
+                }
 
-            final PropertyDescription prop = readPropertyDescription(targetClass, source.getJSONObject("property"));
-            final JSONArray argsAndTypes = source.getJSONArray("args");
-            final Object[] methodArgs = new Object[argsAndTypes.length()];
-            for (int i = 0; i < argsAndTypes.length(); i++) {
-                final JSONArray argPlusType = argsAndTypes.getJSONArray(i);
-                final Object jsonArg = argPlusType.get(0);
-                final String argType = argPlusType.getString(1);
-                methodArgs[i] = convertArgument(jsonArg, argType);
-            }
+                final PropertyDescription prop = readPropertyDescription(targetClass, source.getJSONObject("property"));
+                final JSONArray argsAndTypes = source.getJSONArray("args");
+                final Object[] methodArgs = new Object[argsAndTypes.length()];
+                for (int i = 0; i < argsAndTypes.length(); i++) {
+                    final JSONArray argPlusType = argsAndTypes.getJSONArray(i);
+                    final Object jsonArg = argPlusType.get(0);
+                    final String argType = argPlusType.getString(1);
+                    methodArgs[i] = convertArgument(jsonArg, argType);
+                }
 
-            final Caller mutator = prop.makeMutator(methodArgs);
-            if (null == mutator) {
-                throw new BadInstructionsException("Can't update a read-only property " + prop.name + " (add a mutator to make this work)");
-            }
+                final Caller mutator = prop.makeMutator(methodArgs);
+                if (null == mutator) {
+                    throw new BadInstructionsException("Can't update a read-only property " + prop.name + " (add a mutator to make this work)");
+                }
+
+                return new ViewVisitor.PropertySetVisitor(path, mutator, prop.accessor);
+            } else if (source.getString("change_type").equals("layout")) {
+                final JSONArray args = source.getJSONArray("args");
+                JSONObject layout_info = args.optJSONObject(0);
+                ViewVisitor.LayoutRule params;
+                int verb = layout_info.getInt("verb");
+                if (layout_info.getString("operation").equals("remove")) {
+                    params = new ViewVisitor.LayoutRule(verb, 0);
+                } else if (layout_info.has("anchor_id")) {
```

Comment:
Ah.. Actually, RelativeLayout.TRUE is -1. So, we need to first check if it's "remove", if so, set anchor_id to 0. If not, we want to know if anchor_id is provided. If it's there (indicating it's an anchor layout update), we want to get the anchor_id. If not, we send anchor_id to -1. So, we are having 3 different values for anchor_id here. 

And, on the mixpanel website, I don't want to send "anchor_id=0" for removing, as I don't want to make any assumption of what value Android is using for disabling layouts. Same idea for setting parent layout update, where I can assume RelativeLayout.TRUE to be -1 but I'm not doing it.